src/impy/kinematics.py

```diff
@@ -262,68 +261,51 @@ class EventKinematics(abc.ABC):
         (float) plab     : projectile momentum in lab frame
         (float) elab     : projectile energy in lab frame
         (float) ekin     : projectile kinetic energy in lab frame
-        (float) p1pdg    : PDG ID of the projectile
-        (float) p2pdg    : PDG ID of the target
         (tuple) beam     : Specification as tuple of 4-vectors (np.array)s
-        (tuple) nuc1prop : Projectile nucleus mass & charge (A, Z)
-        (tuple) nuc2prop : Target nucleus mass & charge (A, Z)
+        (tuple) particle1: particle name, PDG ID, or nucleus mass & charge (A, Z)
+                           of the projectile
+        (tuple) particle2: particle name, PDG ID, or nucleus mass & charge (A, Z),
+                           or CompositeTarget of the target
 
     """
 
-    def __init__(*args, **kwargs):
-        raise ValueError(
-            "'EventKinematics' class can not be used directly."
-            + " Use 'FixedTarget' or 'CenterOfMass' instead."
-        )
-
-    def _init(
+    def __init__(
         self,
+        *,
         ecm=None,
         plab=None,
         elab=None,
         ekin=None,
-        p1pdg=None,
-        p2pdg=None,
         beam=None,
-        nuc1_prop=None,
-        nuc2_prop=None,
         particle1=None,
         particle2=None,
```

Comment:
I haven't realised that tag classes that you suggested do not change existing interface. In this case, of course, tag classes are better choice. So I changed the interface to your design.

However, if we add tag classes in `EventKinematics` it will break the existing code that Anatoli uses, because of strings like `(ecm = ...)`. So Anatoli should approve or decline such changes.

Reply:
The tagged float thing looks nice but personally I think it’s a bit overdone :)